eogrow/core/eopatch.py

```diff
@@ -58,28 +59,23 @@ def _prepare_names(self) -> Tuple[bidict, Dict[str, BBox]]:
         """
         bbox_grid = self._area_manager.get_grid(add_bbox_column=True)
 
-        bbox_df = pandas.concat(bbox_grid, ignore_index=True)
-        bbox_df = bbox_df.sort_values(by="index_n")
+        bbox_df: DataFrame = pandas.concat(bbox_grid, ignore_index=True)
 
-        if self._area_manager.has_region_filter():
-            total_patch_num = self._area_manager.get_grid_size(ignore_region_filter=True)
-        else:
-            total_patch_num = len(bbox_df.index)
-
-        prepared_name_to_id_map = self.generate_names(bbox_df, total_patch_num=total_patch_num)
+        prepared_name_to_id_map = self.generate_names(bbox_df)
         prepared_name_to_bbox_map = dict(zip(prepared_name_to_id_map, bbox_df["BBOX"]))
         return prepared_name_to_id_map, prepared_name_to_bbox_map
 
-    def generate_names(self, bbox_dataframe: GeoDataFrame, total_patch_num: Optional[int] = None) -> bidict:
+    def generate_names(self, bbox_dataframe: DataFrame) -> bidict:
         """Method that generates EOPatch names from a dataframe holding bounding boxes. This method can be overridden
         by a method that generates names in a different way.
 
         :param bbox_dataframe: A dataframe with bounding box geometries and information
-        :param total_patch_num: Total number of EOPatches
         :return: A bidirectional dictionary between names and original IDs
         """
-        if total_patch_num is None:
-            total_patch_num = len(bbox_dataframe.index)
+        if len(bbox_dataframe) == 0:
+            return bidict()
```

Comment:
is this needed?

Reply:
Yes, otherwise `bbox_dataframe.total_num.values[0]` line would fail. In grid switching pipeline this method is being called multiple times on various small dataframes. This handles the case if a dataframe would be empty, although in practice it's not a very likely case.